lib/checkclass.cpp

```diff
@@ -2533,18 +2533,29 @@ bool CheckClass::checkConstFunc(const Scope *scope, const Function *func, Member
                 return func && func->isConst();
             };
 
+            auto isConstContainerUsage = [&](const ValueType* vt) -> bool {
+                if (!vt || !vt->container)
+                    return false;
+                const auto yield = vt->container->getYield(end->str());
+                if (((yield == Library::Container::Yield::START_ITERATOR) || (yield == Library::Container::Yield::END_ITERATOR)) &&
+                    (tok1->tokAt(-1)->isComparisonOp() ||
+                     (tok1->tokAt(-1)->isAssignmentOp() && tok1->tokAt(-2)->variable() && Token::Match(tok1->tokAt(-2)->variable()->typeEndToken(), "const_iterator|const_reverse_iterator"))))
```

Comment:
it's not your code but it seems unfortunate to use tokAt here. I'm not sure but wouldnt astparent be better?